Add auto escaping in class PushRequest

ModerRAS opened the pull request:
I found that if I push a text like this: 
```
"" a
```

It will get an exception:
```
NSonic.AssertionException: Expected OK response : ERR invalid_meta_key(?[a"])
```

If I push a text like this:
```
a
a
```
It will get an exception:
```
NSonic.AssertionException: Expected OK response : ERR invalid_format(PUSH <collection> <bucket> <object> "<text>" [LANG(<locale>)]?)
```

I think it needs an auto escaping in NSonic.

spikensbror commented:
Good find! Please complement your PR with unit tests for the described problem-scenarios and everything should be alright.

ModerRAS commented:
I have been so busy recently that I forgot to pay attention to it. I am sorry for that. I added unit tests related to auto escaping and modified the behavior of escaping.

n0099 commented:
@spikensbror May you publish a new NuGet package release with changes in this pr?

n0099 commented:
I've notice that pushing empty string as the text will also trigger the `NSonic.AssertionException: Expected OK response : ERR invalid_format(PUSH <collection> <bucket> <object> "<text>" [LANG(<locale>)]?)`

n0099 commented on NSonic/Impl/Connections/IngestConnection.cs:
```diff
@@ -163,7 +163,7 @@ public PushRequest(string text, string locale)
             }
 
-            public string Text => $"\"{this.text}\"";
+            public string Text => $"\"{this.text.Replace("\n", "\\n").Replace("\"", "\\\"")}\"";
```
The `\` itself also needs a escape and should be done by first:
```suggestion
            public string Text => $"\"{this.text.Replace("\\", "\\\\").Replace("\n", "\\n").Replace("\"", "\\\"")}\"";
```
